uefi/src/table/boot.rs

```diff
@@ -1720,55 +1720,93 @@ impl<'buf> MemoryMap<'buf> {
     #[must_use]
     pub fn entries(&self) -> MemoryMapIter {
         MemoryMapIter {
-            buffer: self.buf,
-            entry_size: self.entry_size,
+            memory_map: self,
             index: 0,
-            len: self.len,
         }
     }
+
+    /// Returns a reference to the [`MemoryDescriptor`] at `index` or `None` if out of bounds.
+    #[inline]
```

Comment:
nit: I think `#[inline]` here is overkill. This code will never be used in any hot path.